internal/rbd/nodeserver.go

```diff
@@ -463,6 +452,46 @@ func (ns *NodeServer) stageTransaction(
 	return transaction, err
 }
 
+// resizeNodeStagePath resizes the device if its encrypted and it also resizes
+// the stagingTargetPath if filesystem needs resize.
+func resizeNodeStagePath(ctx context.Context, transaction *stageTransaction, volID, stagingTargetPath string) error {
+	var err error
+	devicePath := transaction.devicePath
+	if transaction.isEncrypted {
+		mapperFile, mapperPath := util.VolumeMapper(volID)
+		// The volume is encrypted, resize an active mapping
+		err = util.ResizeEncryptedVolume(ctx, mapperFile)
+		if err != nil {
+			log.ErrorLog(ctx, "failed to resize device %s, mapper %s: %w",
+				mapperPath, mapperFile, err)
+
+			return status.Errorf(codes.Internal,
+				"failed to resize device %s, mapper %s: %v", mapperPath, mapperFile, err)
+		}
+		// Use mapper device path for fs resize
+		devicePath = mapperPath
+	}
+	var ok bool
+	resizer := mount.NewResizeFs(utilexec.New())
+	ok, err = resizer.NeedResize(devicePath, stagingTargetPath)
+	if err != nil {
+		return status.Errorf(codes.Internal,
+			"need resize check failed on devicePath %s and staingPath %s, error: %v",
+			devicePath,
+			stagingTargetPath,
+			err)
+	}
+	if ok {
```

Comment:
In most flows we return early in case there is an error or nothing needs to be done. In such a coding style checking for `!ok` on this line and returning fits better. It also reduces the indention of the `resizer.Resize()` call, making the flow a little simpler to follow when just glancing over the code.